### ISSUE TITLE
Errors do not respect the June2018 specs

### ISSUE THREAD

**Issue by bboure** [aws/aws-appsync-community#71]
Since the June 2008 specs, [Errors](https://graphql.github.io/graphql-spec/June2018/#sec-Errors) adds the `extensions` entry. It also discourages the usage of any other additional entries.

> GraphQL services should not provide any additional entries to the error format since they could conflict with additional entries that may be added in future versions of this specification.

AppSync should honor those specs and drop the `data`, `errorInfo` and `errorType` entries from the errors. Insteady, it should implement the `extensions` one.

`$util.error` should probably become `$util.error(String)` and `$util.error(String, Object)`

Where the first argument is the message string, and the second an optional `extensions` map.

**Comment by katzeforest** [comment]
Thanks for the feedback, I will +1 this to the team on your behalf. :)

**Comment by bboure** [comment]
Any news on this?
Thank you

**Comment by steven-chou** [comment]
Any update or progress on this enhancement? A lot of GraphQL client libraries only expose `message`, `locations`, `path` and `extensions` in the errors array in the response. It will be much better to put the existing `errorType`, `errorInfo` and `data` inside the `extensions` map, so it will be easier to access the them from the client side.

**Comment by tombiggs-nz** [comment]
+1 on this. I am using https://pub.dev/packages/graphql_flutter in a project of mine. The standard error response exposes **locations, message, path, and extensions**. It is really annoying that I can not get access to errorType returned from my Appsync API.

Do we have any timeline for this?

**Comment by WhiteHatTux** [comment]
Just to pitch into the people requesting this. I would also really appreciate a nice way to transport my errors to the frontend by following the specification. 
I have the same issue as @tombiggs-nz that I am using graphql_flutter, which correctly follows the specification, but doesn't work with AppSync, as the specification is not followed by appsync. :-(

**Comment by fedecastelli** [comment]
any news about this issue? it has been open for a while and no updates so far.

**Comment by 70nyIT** [comment]
Agree, this is a feature that I expect to find is a product like AppSync

**Comment by shebangcrash** [comment]
Just +1'ing this for our team as well! 
We'd love to be able to leverage the `extensions` field to stay aligned with the formal GraphQL convention :)

**Comment by JordanMRichards** [comment]
+1
